### PR TITLE
Update init containers documentation with 1.6 syntax for YAML files

### PR DESCRIPTION
**Pull request by crimsonfaith91**
addressing https://github.com/kubernetes/kubernetes.github.io/issues/3261

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3572)

### CODE REVIEW

File: docs/concepts/workloads/pods/init-containers.md [hunk 1]
```diff
@@ -98,11 +98,11 @@ and the [Production Pods guide](/docs/user-guide/production-pods.md#handling-ini
 
 ### Init Containers in use
 
-The following yaml file outlines a simple Pod which has two Init Containers.
+The following yaml file for Kubernetes version 1.5 outlines a simple Pod which has two Init Containers.
 The first waits for `myservice` and the second waits for `mydb`. Once both
 containers complete the Pod will begin.
 
-```yaml
+```yaml for version 1.5
```
Comment (foxish) [inline docs/concepts/workloads/pods/init-containers.md]: The yaml section specified this way is special. It shouldn't be modified.

File: docs/concepts/workloads/pods/init-containers.md [hunk 2]
```diff
@@ -129,6 +129,30 @@ spec:
     command: ['sh', '-c', 'echo The app is running! && sleep 3600']
 ```
 
+There is a slight change of syntax in Kubernetes version 1.6.
+We move declaration of the init containers to `spec`:
```
Comment (foxish) [inline docs/concepts/workloads/pods/init-containers.md]: In Kubernetes 1.6, init-containers were made a field in the API. The beta annotation is still respected but will be deprecated in future releases.

File: docs/concepts/workloads/pods/init-containers.md [hunk 3]
```diff
@@ -129,6 +129,30 @@ spec:
...
+
+```yaml for version 1.6
```
Comment (foxish) [inline docs/concepts/workloads/pods/init-containers.md]: should just be ```yaml

File: docs/concepts/workloads/pods/init-containers.md (hunk 1 above)
Comment (crimsonfaith91) [inline docs/concepts/workloads/pods/init-containers.md]: noted!

File: docs/concepts/workloads/pods/init-containers.md (hunk 2 above)
Comment (crimsonfaith91) [inline docs/concepts/workloads/pods/init-containers.md]: I will add your comment into the doc directly.

File: docs/concepts/workloads/pods/init-containers.md (hunk 3 above)
Comment (crimsonfaith91) [inline docs/concepts/workloads/pods/init-containers.md]: yup!

**Comment by chenopis** [comment]
@foxish So is that a Tech LGTM?

**Comment by foxish** [comment]
/lgtm
@chenopis Yes.

File: docs/concepts/workloads/pods/init-containers.md [hunk 4]
```diff
@@ -129,6 +129,55 @@ spec:
     command: ['sh', '-c', 'echo The app is running! && sleep 3600']
 ```
 
+There is a slight change of syntax in Kubernetes 1.6. We move declaration of the init containers to `spec`:
```
Comment (jberkus) [inline docs/concepts/workloads/pods/init-containers.md]: replace: "There is a new syntax in Kubernetes 1.6, although the old annotation syntax still works.  We have moved the declaration of init containers to `spec`:"

**Comment by crimsonfaith91** [comment]
@chenopis Thanks for the help in rephrasing the description! :)

**Comment by chenopis** [comment]
@crimsonfaith91 Was just implementing the suggestion from @jberkus =)

**Pull request merged**